src/lib/internal/arrays.ts

```diff
@@ -0,0 +1,8 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function arraysAreEqual(arr1: any[], arr2: any[]): boolean {
```

Comment:
This can be changed to use `unknown[]` instead:
```suggestion
export function arraysAreEqual(arr1: unknown[], arr2: unknown[]): boolean {
```

We can also use a generic instead to ensure that the types are the same too:
```suggestion
export function arraysAreEqual<T extends Array<unknown>>(arr1: T, arr2: T): boolean {
```